stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -1589,15 +1589,14 @@ extension ${Self} {
 %   ThatBuiltinName = self_ty.builtin_name
 %   sign = 's' if self_ty.is_signed else 'u'
 %   srcBits = self_ty.bits
-%   maxExponent = (1 << (ExponentBitCount - 1)) - 1
-%   maxBitLength = SignificandBitCount + 1 # implicit leading one
-%   if self_ty.is_signed:
-%     # The magnitude of the minimum value of a signed integer type is always
-%     # representable in floating point (assuming a large enough exponent)
-%     # because only one bit is set. All other possible magnitudes can be
-%     # represented using one bit less than the original source type.
-%     maxBitLength += 1
-%   end
+%
+%   max_required_exponent = srcBits - 1
```

Comment:
All the other names use camel case. Is there a reason to switch? I don't have a preference (if anything I prefer using underscores) but consistency makes things easier to read.

Reply:
Not all the other names use camel case, and from what I can glean, there is an actual method to the madness:

Variables that are emitted in some way in the generated Swift source adopt Swift naming conventions which reflect their primary use case in the generated source. In other words, those Python variables that represent types (`ThatBuiltinName`) are capitalized as though a Swift type, and those that represent instance properties (`srcBits`) adopt camelCase.

Notice, however, that other variables defined in Python (`self_ty.is_signed`) use snake_case. Here, none of these variables are emitted into the Swift source code; therefore, after some initial waffling, I have made them snake_case.